exe/Main.hs

```diff
@@ -175,7 +176,7 @@ showEvent lock (EventFileDiagnostics (toNormalizedFilePath -> file) diags) =
 showEvent lock e = withLock lock $ print e
 
 
-cradleToSession :: Cradle -> IO HscEnvEq
+cradleToSession :: Cradle Void -> IO HscEnvEq
```

Comment:
Could this equally be `Cradle a`, so it was polymorphic in the type of Cradle it receives? That seems slightly simpler.

Reply:
Yeah sure, should not be a problem.